wallet/src/service/Offers.ts

```diff
@@ -257,15 +284,41 @@ export const getOfferService = (
   };
 
   const watchPendingOffers = async (brandToPurse: Map<Brand, PurseInfo>) => {
-    const makeProposalEntriesDisplayable = (proposalEntries: {
+    const makeProposalEntriesDisplayable = async (proposalEntries: {
       [key: string]: Amount;
     }) =>
       Object.fromEntries(
-        Object.entries(proposalEntries).map(([kw, entry]) => {
-          const pursePetname = brandToPurse.get(entry.brand)?.pursePetname;
-          const value = String(entry.value);
-          return [kw, { pursePetname, value }];
-        }),
+        await Promise.all(
```

Comment:
this looks like an opportunity to simplify with `objectMap` and `deeplyFulfilledObject`. I think with that you would need no `await`

Reply:
check out [04ae1c6](https://github.com/Agoric/wallet-app/pull/125/commits/04ae1c6eb875d01465b66ed2c1c94f5188898fd6)